core/src/plan/join.rs

```diff
@@ -898,69 +663,20 @@ mod tests {
             );
         ";
         let actual = plan_join(&storage, sql);
-        let expected = {
-            let join_executor = JoinExecutor::Hash {
-                key_expr: expr("u4.id"),
-                value_expr: expr("u3.id + u1.id"),
-                where_clause: None,
-            };
-
-            let join_subquery = Query {
-                body: SetExpr::Select(Box::new(Select {
-                    projection: vec![SelectItem::Wildcard],
-                    from: TableWithJoins {
-                        relation: table_factor("Player", Some("u3")),
-                        joins: vec![Join {
-                            relation: table_factor("Player", Some("u4")),
-                            join_operator: inner(None),
-                            join_executor,
-                        }],
-                    },
-                    selection: None,
-                    group_by: Vec::new(),
-                    having: None,
-                })),
-                order_by: vec![],
-                limit: None,
-                offset: None,
-            };
-
-            let subquery = Query {
-                body: SetExpr::Select(Box::new(Select {
-                    projection: vec![SelectItem::Wildcard],
-                    from: TableWithJoins {
-                        relation: table_factor("Player", Some("u2")),
-                        joins: Vec::new(),
-                    },
-                    selection: Some(Expr::BinaryOp {
-                        left: Box::new(expr("u2.id")),
-                        op: BinaryOperator::Eq,
-                        right: Box::new(Expr::Subquery(Box::new(join_subquery))),
-                    }),
-                    group_by: Vec::new(),
-                    having: None,
-                })),
-                order_by: vec![],
-                limit: None,
-                offset: None,
-            };
-
-            select(Select {
-                projection: vec![SelectItem::Wildcard],
-                from: TableWithJoins {
-                    relation: table_factor("Player", Some("u1")),
-                    joins: Vec::new(),
-                },
-                selection: Some(Expr::BinaryOp {
-                    left: Box::new(expr("u1.id")),
-                    op: BinaryOperator::Eq,
-                    right: Box::new(Expr::Subquery(Box::new(subquery))),
-                }),
-                group_by: Vec::new(),
-                having: None,
-            })
-        };
-        assert_eq!(actual, expected, "hash join in nested subquery:\n{sql}");
+        let expected = table("Player").alias_as("u1").select().filter(
```

Comment:
I am so impressed by the increase in readability through the query builder function that has such a long ast.